flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraph.java

```diff
@@ -79,6 +81,11 @@ private void handleExecutionGraphCreation(
                     throwable);
             context.goToFinished(context.getArchivedExecutionGraph(JobStatus.FAILED, throwable));
         } else {
```

Comment:
OT: A sanity check would make sense here
```suggestion
        } else {
            Objects.requireNonNull(executionGraphWithVertexParallelism, "...");
```